russh-keys/src/format/pkcs8.rs

```diff
@@ -164,6 +165,59 @@ fn read_key_v1(reader: &mut BERReaderSeq) -> Result<key::KeyPair, Error> {
     }
 }
 
+#[cfg(not(feature = "openssl"))]
+fn write_key_v0(writer: &mut yasna::DERWriterSeq, key: &RsaPrivateKey) {
+    // Construct DER sequence
+    writer.next().write_u32(0); // Version
+    writer.next().write_sequence(|writer| {
+        writer.next().write_oid(&ObjectIdentifier::from_slice(RSA));
+        writer.next().write_null();
+    });
+
+    // Encode key data
+    let bytes = yasna::construct_der(|writer| {
+        let _ = writer.write_sequence(|writer| -> Result<(), Error> {
+            writer.next().write_u32(0); // Version
+            use num_bigint::BigUint;
+
+            writer
+                .next()
+                .write_biguint(&BigUint::from_bytes_be(&key.n().to_bytes_be()));
+            writer
+                .next()
+                .write_biguint(&BigUint::from_bytes_be(&key.e().to_bytes_be()));
+            writer
+                .next()
+                .write_biguint(&BigUint::from_bytes_be(&key.d().to_bytes_be()));
+
+            let primes = key.primes();
+
+            writer.next().write_biguint(&BigUint::from_bytes_be(
+                &primes.get(0).ok_or(Error::IndexOutOfBounds)?.to_bytes_be(),
+            ));
+            writer.next().write_biguint(&BigUint::from_bytes_be(
+                &primes.get(1).ok_or(Error::IndexOutOfBounds)?.to_bytes_be(),
+            ));
+
+            if let (Some(dp), Some(dq), Some(iqmp)) = (key.dp(), key.dq(), key.crt_coefficient()) {
```

Comment:
These fields are not optional. See: https://datatracker.ietf.org/doc/html/rfc8017#appendix-A.1.2
I think you could use `key.precompute()` to make sure these are available, but that requires you to hold a mutable ref or make a copy here.

Reply:
Done ✅